kv/txn_coord_sender_test.go

```diff
@@ -696,34 +699,44 @@ func TestTxnCoordSenderSingleRoundtripTxn(t *testing.T) {
 func TestTxnCoordSenderErrorWithIntent(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	stopper := stop.NewStopper()
+	defer stopper.Stop()
 	manual := hlc.NewManualClock(0)
 	clock := hlc.NewClock(manual.UnixNano)
 	clock.SetMaxOffset(20)
 
-	ts := NewTxnCoordSender(senderFn(func(_ context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
-		txn := ba.Txn.Clone()
-		txn.Writing = true
-		pErr := roachpb.NewError(roachpb.NewTransactionRetryError())
-		pErr.SetTxn(&txn)
-		return nil, pErr
-	}), clock, false, tracing.NewTracer(), stopper, NewTxnMetrics(metric.NewRegistry()))
-	defer stopper.Stop()
-
-	var ba roachpb.BatchRequest
-	key := roachpb.Key("test")
-	ba.Add(&roachpb.BeginTransactionRequest{Span: roachpb.Span{Key: key}})
-	ba.Add(&roachpb.PutRequest{Span: roachpb.Span{Key: key}})
-	ba.Add(&roachpb.EndTransactionRequest{})
-	ba.Txn = &roachpb.Transaction{Name: "test"}
-	if _, pErr := ts.Send(context.Background(), ba); !testutils.IsPError(pErr, "retry txn") {
-		t.Fatalf("unexpected error: %v", pErr)
+	testPErrs := []*roachpb.Error{
+		roachpb.NewError(roachpb.NewTransactionRetryError()),
+		roachpb.NewError(roachpb.NewTransactionAbortedError()),
+		roachpb.NewError(roachpb.NewTransactionPushError(roachpb.Transaction{
+			TxnMeta: roachpb.TxnMeta{
+				ID: uuid.NewV4(),
+			}})),
+		roachpb.NewErrorf("testError"),
 	}
+	for i, testPErr := range testPErrs {
+		ts := NewTxnCoordSender(senderFn(func(_ context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
```

Comment:
could you wrap the inside of this loop in a `func() { /* ... */ }()` so that the defers run earlier?